x/ibc/05-port/types/module.go

```diff
@@ -61,7 +61,7 @@ type IBCModule interface {
 	OnRecvPacket(
 		ctx sdk.Context,
 		packet channeltypes.Packet,
-	) (*sdk.Result, error)
+	) (*sdk.Result, []byte, error)
```

Comment:
I decided to modify the callback interface because
* it is more clear to the application developer the contract required to fulfill this function securely
* using `sdk.Result.Data` would be messy because you would need to require some specified format for the ack such as being prefixed by some string ie `[]byte("acknowledgement")+ack.GetBytes()`

Reply:
Agreed, I think this is cleaner for the time being.